### PR TITLE
Disallow Ubuntu arm32 non-PR jobs in non-master branches

### PR DESCRIPTION
**Pull request by BruceForstall**

### CODE REVIEW

**Comment by BruceForstall** [comment]
@dotnet-bot test ci please

**Comment by BruceForstall** [comment]
This is to stop push and cron Ubuntu arm32 jobs in dev/unix_test_workflow, but should get pushed to master as well.

File: netci.groovy [hunk 1]
```diff
@@ -1074,13 +1075,9 @@ def static getJobName(def configuration, def architecture, def os, def scenario,
...
-    // Limited Windows ARM64 hardware is restricted for non-PR triggers to certain branches.
-    if (os == 'Windows_NT') {
-        if ((architecture == 'arm64') || (architecture == 'arm') || (architecture == 'armlb')) {
-            if (!(branch in Constants.WindowsArm64Branches)) {
-                return
-            }
-        }
+    // Limited hardware is restricted for non-PR triggers to certain branches.
+    if (jobRequiresLimitedHardware(architecture, os) && (!(branch in Constants.LimitedHardwareBranches))) {
+        return
```
Comment (echesakovMSFT) [inline netci.groovy]: nit: shall we create `branchCanUseLimitedHardware(branch)` predicate and do something like?
```
if (jobRequiresLimitedHardware(architecture, os) && !branchCanUseLimitedHardware(branch))
    return
```

Comment (BruceForstall) [reply netci.groovy]: It's a good idea; I'll try to remember that for a future change

**Pull request merged**